// @generated/type-graphql/resolvers/relations/JobApplication/JobApplicationRelationsResolver.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.JobApplicationRelationsResolver = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = (0, tslib_1.__importStar)(require("type-graphql"));
const Intern_1 = require("../../../models/Intern");
const Job_1 = require("../../../models/Job");
const JobApplication_1 = require("../../../models/JobApplication");
const helpers_1 = require("../../../helpers");
let JobApplicationRelationsResolver = class JobApplicationRelationsResolver {
    async intern(jobApplication, ctx) {
        return (0, helpers_1.getPrismaFromContext)(ctx).jobApplication.findUnique({
            where: {
                id: jobApplication.id,
            },
        }).intern({});
    }
    async job(jobApplication, ctx) {
        return (0, helpers_1.getPrismaFromContext)(ctx).jobApplication.findUnique({
            where: {
                id: jobApplication.id,
            },
        }).job({});
    }
};
(0, tslib_1.__decorate)([
    TypeGraphQL.FieldResolver(_type => Intern_1.Intern, {
        nullable: false
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Root()),
    (0, tslib_1.__param)(1, TypeGraphQL.Ctx()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [JobApplication_1.JobApplication, Object]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], JobApplicationRelationsResolver.prototype, "intern", null);
(0, tslib_1.__decorate)([
    TypeGraphQL.FieldResolver(_type => Job_1.Job, {
        nullable: false
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Root()),
    (0, tslib_1.__param)(1, TypeGraphQL.Ctx()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [JobApplication_1.JobApplication, Object]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], JobApplicationRelationsResolver.prototype, "job", null);
JobApplicationRelationsResolver = (0, tslib_1.__decorate)([
    TypeGraphQL.Resolver(_of => JobApplication_1.JobApplication)
], JobApplicationRelationsResolver);
exports.JobApplicationRelationsResolver = JobApplicationRelationsResolver;
