// @generated/type-graphql/resolvers/crud/Job/args/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpsertJobArgs = exports.UpdateManyJobArgs = exports.UpdateJobArgs = exports.GroupByJobArgs = exports.FindUniqueJobArgs = exports.FindManyJobArgs = exports.FindFirstJobArgs = exports.DeleteManyJobArgs = exports.DeleteJobArgs = exports.CreateManyJobArgs = exports.CreateJobArgs = exports.AggregateJobArgs = void 0;
var AggregateJobArgs_1 = require("./AggregateJobArgs");
Object.defineProperty(exports, "AggregateJobArgs", { enumerable: true, get: function () { return AggregateJobArgs_1.AggregateJobArgs; } });
var CreateJobArgs_1 = require("./CreateJobArgs");
Object.defineProperty(exports, "CreateJobArgs", { enumerable: true, get: function () { return CreateJobArgs_1.CreateJobArgs; } });
var CreateManyJobArgs_1 = require("./CreateManyJobArgs");
Object.defineProperty(exports, "CreateManyJobArgs", { enumerable: true, get: function () { return CreateManyJobArgs_1.CreateManyJobArgs; } });
var DeleteJobArgs_1 = require("./DeleteJobArgs");
Object.defineProperty(exports, "DeleteJobArgs", { enumerable: true, get: function () { return DeleteJobArgs_1.DeleteJobArgs; } });
var DeleteManyJobArgs_1 = require("./DeleteManyJobArgs");
Object.defineProperty(exports, "DeleteManyJobArgs", { enumerable: true, get: function () { return DeleteManyJobArgs_1.DeleteManyJobArgs; } });
var FindFirstJobArgs_1 = require("./FindFirstJobArgs");
Object.defineProperty(exports, "FindFirstJobArgs", { enumerable: true, get: function () { return FindFirstJobArgs_1.FindFirstJobArgs; } });
var FindManyJobArgs_1 = require("./FindManyJobArgs");
Object.defineProperty(exports, "FindManyJobArgs", { enumerable: true, get: function () { return FindManyJobArgs_1.FindManyJobArgs; } });
var FindUniqueJobArgs_1 = require("./FindUniqueJobArgs");
Object.defineProperty(exports, "FindUniqueJobArgs", { enumerable: true, get: function () { return FindUniqueJobArgs_1.FindUniqueJobArgs; } });
var GroupByJobArgs_1 = require("./GroupByJobArgs");
Object.defineProperty(exports, "GroupByJobArgs", { enumerable: true, get: function () { return GroupByJobArgs_1.GroupByJobArgs; } });
var UpdateJobArgs_1 = require("./UpdateJobArgs");
Object.defineProperty(exports, "UpdateJobArgs", { enumerable: true, get: function () { return UpdateJobArgs_1.UpdateJobArgs; } });
var UpdateManyJobArgs_1 = require("./UpdateManyJobArgs");
Object.defineProperty(exports, "UpdateManyJobArgs", { enumerable: true, get: function () { return UpdateManyJobArgs_1.UpdateManyJobArgs; } });
var UpsertJobArgs_1 = require("./UpsertJobArgs");
Object.defineProperty(exports, "UpsertJobArgs", { enumerable: true, get: function () { return UpsertJobArgs_1.UpsertJobArgs; } });
