// @generated/type-graphql/enums/CompanyScalarFieldEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CompanyScalarFieldEnum = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = (0, tslib_1.__importStar)(require("type-graphql"));
var CompanyScalarFieldEnum;
(function (CompanyScalarFieldEnum) {
    CompanyScalarFieldEnum["id"] = "id";
    CompanyScalarFieldEnum["userId"] = "userId";
    CompanyScalarFieldEnum["name"] = "name";
    CompanyScalarFieldEnum["description"] = "description";
    CompanyScalarFieldEnum["logo"] = "logo";
    CompanyScalarFieldEnum["website"] = "website";
    CompanyScalarFieldEnum["CIN"] = "CIN";
    CompanyScalarFieldEnum["email"] = "email";
    CompanyScalarFieldEnum["createdAt"] = "createdAt";
    CompanyScalarFieldEnum["updatedAt"] = "updatedAt";
})(CompanyScalarFieldEnum = exports.CompanyScalarFieldEnum || (exports.CompanyScalarFieldEnum = {}));
TypeGraphQL.registerEnumType(CompanyScalarFieldEnum, {
    name: "CompanyScalarFieldEnum",
    description: undefined,
});
