// @generated/type-graphql/enums/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserScalarFieldEnum = exports.Tag = exports.Status = exports.SortOrder = exports.Role = exports.QueryMode = exports.JobType = exports.JobScalarFieldEnum = exports.JobApplicationScalarFieldEnum = exports.InternScalarFieldEnum = exports.CompanyScalarFieldEnum = void 0;
var CompanyScalarFieldEnum_1 = require("./CompanyScalarFieldEnum");
Object.defineProperty(exports, "CompanyScalarFieldEnum", { enumerable: true, get: function () { return CompanyScalarFieldEnum_1.CompanyScalarFieldEnum; } });
var InternScalarFieldEnum_1 = require("./InternScalarFieldEnum");
Object.defineProperty(exports, "InternScalarFieldEnum", { enumerable: true, get: function () { return InternScalarFieldEnum_1.InternScalarFieldEnum; } });
var JobApplicationScalarFieldEnum_1 = require("./JobApplicationScalarFieldEnum");
Object.defineProperty(exports, "JobApplicationScalarFieldEnum", { enumerable: true, get: function () { return JobApplicationScalarFieldEnum_1.JobApplicationScalarFieldEnum; } });
var JobScalarFieldEnum_1 = require("./JobScalarFieldEnum");
Object.defineProperty(exports, "JobScalarFieldEnum", { enumerable: true, get: function () { return JobScalarFieldEnum_1.JobScalarFieldEnum; } });
var JobType_1 = require("./JobType");
Object.defineProperty(exports, "JobType", { enumerable: true, get: function () { return JobType_1.JobType; } });
var QueryMode_1 = require("./QueryMode");
Object.defineProperty(exports, "QueryMode", { enumerable: true, get: function () { return QueryMode_1.QueryMode; } });
var Role_1 = require("./Role");
Object.defineProperty(exports, "Role", { enumerable: true, get: function () { return Role_1.Role; } });
var SortOrder_1 = require("./SortOrder");
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return SortOrder_1.SortOrder; } });
var Status_1 = require("./Status");
Object.defineProperty(exports, "Status", { enumerable: true, get: function () { return Status_1.Status; } });
var Tag_1 = require("./Tag");
Object.defineProperty(exports, "Tag", { enumerable: true, get: function () { return Tag_1.Tag; } });
var UserScalarFieldEnum_1 = require("./UserScalarFieldEnum");
Object.defineProperty(exports, "UserScalarFieldEnum", { enumerable: true, get: function () { return UserScalarFieldEnum_1.UserScalarFieldEnum; } });
