// @generated/type-graphql/enums/JobScalarFieldEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.JobScalarFieldEnum = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = (0, tslib_1.__importStar)(require("type-graphql"));
var JobScalarFieldEnum;
(function (JobScalarFieldEnum) {
    JobScalarFieldEnum["id"] = "id";
    JobScalarFieldEnum["position"] = "position";
    JobScalarFieldEnum["description"] = "description";
    JobScalarFieldEnum["jobType"] = "jobType";
    JobScalarFieldEnum["skillsRequired"] = "skillsRequired";
    JobScalarFieldEnum["numOfOpenings"] = "numOfOpenings";
    JobScalarFieldEnum["duration"] = "duration";
    JobScalarFieldEnum["postedAt"] = "postedAt";
    JobScalarFieldEnum["location"] = "location";
    JobScalarFieldEnum["salary"] = "salary";
    JobScalarFieldEnum["industry"] = "industry";
    JobScalarFieldEnum["closed"] = "closed";
    JobScalarFieldEnum["companyId"] = "companyId";
    JobScalarFieldEnum["updatedAt"] = "updatedAt";
})(JobScalarFieldEnum = exports.JobScalarFieldEnum || (exports.JobScalarFieldEnum = {}));
TypeGraphQL.registerEnumType(JobScalarFieldEnum, {
    name: "JobScalarFieldEnum",
    description: undefined,
});
